lib/rules/no-useless-rename.js

```diff
@@ -87,26 +91,21 @@ module.exports = {
                 return;
             }
 
-            const properties = node.properties;
-
-            for (let i = 0; i < properties.length; i++) {
-                if (properties[i].shorthand) {
-                    continue;
-                }
+            for (const property of node.properties) {
 
                 /**
-                 * If an ObjectPattern property is computed, we have no idea
-                 * if a rename is useless or not. If an ObjectPattern property
-                 * lacks a key, it is likely an ExperimentalRestProperty and
-                 * so there is no "renaming" occurring here.
+                 * Properties using shorthand syntax and rest elements can not be renamed.
+                 * If the property is computed, we have no idea if a rename is useless or not.
                  */
-                if (properties[i].computed || !properties[i].key) {
+                if (property.shorthand || property.type === "RestElement" || property.computed) {
                     continue;
                 }
 
-                if (properties[i].key.type === "Identifier" && properties[i].key.name === properties[i].value.name ||
-                        properties[i].key.type === "Literal" && properties[i].key.value === properties[i].value.name) {
-                    reportError(properties[i], properties[i].key, properties[i].value, "Destructuring assignment");
+                const key = (property.key.type === "Identifier" && property.key.name) || (property.key.type === "Literal" && property.key.value);
```

Comment:
Some change here seems to break! 

CI Error: https://circleci.com/gh/nuxt/nuxt.js/55219?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link

Failing line: https://github.com/nuxt/nuxt.js/blob/13331a02b4d5ef37932f42a845ca1754fb834243/packages/webpack/src/builder.js#L169

(just informational! I'll try to reproduce it better and make PR or Issue)

Reply:
Please track this issue: #12335 .